skymp5-server/cpp/server_guest_lib/PapyrusActor.cpp

```diff
@@ -178,3 +178,38 @@ VarValue PapyrusActor::SetDontMove(VarValue self,
   }
   return VarValue::None();
 }
+
+VarValue PapyrusActor::WornHasKeyword(VarValue self,
+                                      const std::vector<VarValue>& arguments)
+{
+  if (auto actor = GetFormPtr<MpActor>(self)) {
+    if (arguments.size() < 1) {
+      throw std::runtime_error(
+        "Actor.WornHasKeyword requires at least one argument");
+    }
+
+    const auto& keywordRec = GetRecordPtr(arguments[0]);
+    if (!keywordRec.rec) {
+      spdlog::error("Actor.WornHasKeyword - invalid keyword form");
+      return VarValue(false);
+    }
+
+    std::vector<Inventory::Entry> entries = actor->GetEquipment().inv.entries;
+    for (size_t i = 0; i < entries.size(); i++) {
+      WorldState* worldState = compatibilityPolicy->GetWorldState();
+      const auto& keywordIds =
+        worldState->GetEspm()
+          .GetBrowser()
+          .LookupById(entries[i].baseId)
+          .rec->GetKeywordIds(worldState->GetEspmCache());
+      for (auto rawId : keywordIds) {
+        if (keywordRec.ToGlobalId(rawId) ==
```

Comment:
auto lookupRes = worldState->GetEspm()
          .GetBrowser()
          .LookupById(entries[i].baseId);

lookupRes.toGlobalId

Reply:
xxxx.toGlobalId will only work for rawIds from xxxx variable